### PR TITLE
C#: Include all (legacy) nuget restored folders in standalone references

### PR DESCRIPTION
**Pull request by tamasvajk**
This PR 
- adds all dlls from nuget packages that have been restored with `nuget.exe restore` to the reference list of standalone,
- except the ones that are in `/tools/` folder (see details [here](https://learn.microsoft.com/en-us/nuget/create-packages/creating-a-package#solution-level-packages)).

### CODE REVIEW

**Comment by tamasvajk** [comment]
DCA results look almost good. There are many new compiler errors on `AngleSharp/AngleSharp`. It looks like nuget 2 allowed adding a `tools` folder that can contain assemblies which are not references to the application. We could/should exclude those.

**Review by michaelnebel — APPROVED** [review]
Looks good to me!

**Comment by tamasvajk** [comment]
DCA looks good: there are 328 compilation errors fixed, and there's only one project with increased error count.

File: csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/DependencyManager.cs [hunk 1]
```diff
@@ -89,8 +91,21 @@ public DependencyManager(string srcDir, IDependencyOptions options, ILogger logg
             {
                 try
                 {
-                    var nuget = new NugetPackages(sourceDir.FullName, packageDirectory, progressMonitor);
+                    var nuget = new NugetPackages(sourceDir.FullName, legacyPackageDirectory, progressMonitor);
```
[the remark this answers is not in the document]
Comment (tamasvajk) [reply csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/DependencyManager.cs:94]: I don't know if that's possible with the old csproj structure. I'm under the assumption that .net framework packages will not be downloaded when nuget.exe is being used.

[the remark this answers is not in the document]
Comment (michaelnebel) [reply csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/DependencyManager.cs:94]: Great!

**Review by michaelnebel — APPROVED** [review]
LGTM 👍

**Pull request merged**